Dockerfile

```diff
@@ -64,10 +64,23 @@ RUN apk add --update --no-cache libpq tzdata yaml ${EXTRA_PACKAGES} && \
     cp /usr/share/zoneinfo/Europe/London /etc/localtime && \
     echo "Europe/London" > /etc/timezone
 
+# Create non-root user and group with specific UIDs/GIDs
+RUN addgroup -S appgroup -g 20001 && adduser -S appuser -G appgroup -u 10001
+
 # Copy files generated in the builder image
 COPY --from=builder /app /app
 COPY --from=builder /usr/local/bundle/ /usr/local/bundle/
 
+# Change ownership of the app directory to the non-root user
+RUN chown -R appuser:appgroup /app
+
+# Make everything read-only (with execute for directories)
+RUN chmod -R a-w /app && \
+    find /app -type d -exec chmod +x {} \;
+
+# Grant write access only to directories that need it
+RUN chmod -R u+w /app/tmp /app/public/api/docs /app/log
```

Comment:
As we run a chown on everything below /app, the chmod commands aren't going to give us any security improvement i.e. if you connect to the pod you can change it. The ask here is can we restrict the directories we need to chown instead of recursively chown /app e.g. just chown /app/log, /app/tmp, ...